index.html

```diff
@@ -112,19 +84,76 @@
     .style("text-align", "center")
     .style("font-family", "monospace")
     .style("width", "400px")
-    .text("")
+    .html("")
+
+nodes.forEach(function(node) {
+    node.x = width
+    node.y = height
+})
+
+// creating a force between the links
+var linkForce = d3.forceLink()
+    .id(function (link) { return link.id })
+    .distance(getDistance)
+    .iterations(1)
+
+function getDistance(link) {
+  return link.source.radius + link.target.radius + PERSPECT_LINK
+}
+
+// returns the color of the node based on its level
+function getNodeColor(node) {
+    return node.level === 1 ? '#29f2ba' : '#5da5ef'
+}
+
+// returns the radius of the node based on the number
+// of children it has
+function getRadius(node) {
+    if (node.level === 1) {
+        return node.radius
+    } else {
+        return 14
+    }
+}
+
+var modulePosition = {
+  "3": {x: 0, y: 0},
+  "2": {x: 0, y: 0},
+  "1": {x: 0, y: 0},
+  "0": {x: 0, y: 0}
+}
+
+//Make the x-position equal to the x-position specified in the module positioning object or, if not in
+//the hash, then set it to 250
+var forceX = d3.forceX(function (d) {return modulePosition[d.module] ? modulePosition[d.module].x : 250})
```

Comment:
I was super confused what the whole module position was doing because it always seems to be 0 however it seems to make a difference whether it's there or not. We definitely need the forceX and forceY for centering and when it's replaced with just `d3.forceX(0)` and `d3.forceY(0)` the spacing gets a lot tighter. It's kind of hard to tell because the positions are still randomized but there's a noticeable difference that's not optimal.